### PR TITLE
[Merged by Bors] - chore: lighten tactics

### PR DESCRIPTION
**Pull request by hrmacbeth**
Change some tactic uses to more lightweight ones (but which still automate just as much): `nlinarith` to `positivity`/`linarith`/`gcongr`, `linarith` to `norm_num`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/PSeries.lean [hunk 1]
```diff
@@ -392,8 +392,7 @@ theorem sum_Ioc_inv_sq_le_sub {k n : ℕ} (hk : k ≠ 0) (h : k ≤ n) :
   · ring_nf
     rw [add_comm]
     exact B.le
-  · -- Porting note: was `nlinarith`
-    positivity
+  · positivity
```
Comment (hrmacbeth) [inline Mathlib/Analysis/PSeries.lean:395]: Removing this porting note: `nlinarith` works here now, it's just a worse proof!

**Comment by grunweg** [comment]
The changes itself look fine to me; thanks for starting this. Please fix the merge conflict.
I presume your motivation is performance - can you benchmark this PR once it builds?

**Comment by hrmacbeth** [comment]
Yes, I believe these are all changes which improve performance (of course, microscopically in the scale of the library -- I'm not sure our profiling tools would detect it) -- while also being "uncontroversial" in the sense that there is no loss of automation (the new tactics run without user input).

But there is no direct motivation behind the PR, these are just things I happened to notice while looking at the files for other reasons.

**Comment by grunweg** [comment]
Thanks for explaining. I'll wait to see if benchmarking turns up anything surprising - but this PR seems non-harmful regardless, so as I already reviewed it, I'd say we can merge it as well.
maintainer merge

**Pull request closed without merging**